gennet-module/gennet.py

```diff
@@ -265,103 +247,79 @@ def generate_and_write_files(dirname, num_topics, num_subnets, node_type_distrib
-    return num_partitions
-
-
-# sanity check :  num_subnets < num_nodes
-def _num_subnets_callback(ctx: typer, Context, num_subnets: int):
-    num_nodes = ctx.params["num_nodes"]
-    if num_subnets > num_nodes:
-        raise ValueError(
-            f"num_subnets must be <= num_nodes: num_subnets={num_subnets}, num_nodes={1}")
-    if num_subnets == -1:
-        num_subnets = num_nodes
-    return num_subnets
-
-
 # sanity check : valid json with "gennet" config
 def conf_callback(ctx: typer.Context, param: typer.CallbackParam, cfile: str):
     if cfile:
         typer.echo(f"Loading config file: {cfile.split('/')[-1]}")
+        ctx.default_map = ctx.default_map or {}  # Init the default map
         try:
             with open(cfile, 'r') as f:  # Load config file
                 conf = json.load(f)
                 if "gennet" not in conf:
                     print(f"Gennet configuration not found in {cfile}. Skipping topology generation.")
                     sys.exit(1)
+                if "general" in conf and "prng_seed" in conf["general"]:
+                    conf["gennet"]["prng_seed"] = conf["general"]["prng_seed"]
                 # TODO : type-check and sanity-check the config.json
                 #print(conf)
+            ctx.default_map.update(conf["gennet"])  # Merge config and default_map
         except Exception as ex:
             raise typer.BadParameter(str(ex))
     return cfile
 
 
-# methods to merge cli values, config.json and default values
-def test_and_set_int(cli_val, file_val, conf, module="gennet"):
-    if cli_val != INT_NONE:
-        return cli_val
-    if cli_val == INT_NONE and module in conf and file_val in conf[module]:
-        return conf[module][file_val]
-    return defaults[file_val]
+# sanity check : num_partitions == 1
+def _num_partitions_callback(num_partitions: int):
+    if num_partitions > 1:
+        raise ValueError(
+            f"--num-partitions {num_partitions}, Sorry, we do not yet support partitions")
+    return num_partitions
 
 
-def test_and_set_str(cli_val, file_val, conf, module="gennet"):
-    if cli_val != STR_NONE:
-        return cli_val
-    if cli_val == STR_NONE and module in conf and file_val in conf[module]:
-        return conf[module][file_val]
-    return defaults[file_val]
+# sanity check :  num_subnets < num_nodes
+def _num_subnets_callback(ctx: typer, Context, num_subnets: int):
+    num_nodes = ctx.params["num_nodes"]
+    if num_subnets == -1:
+        num_subnets = num_nodes
+    if num_subnets > num_nodes:
+        raise ValueError(
+            f"num_subnets must be <= num_nodes: num_subnets={num_subnets}, num_nodes={1}")
+    return num_subnets
 
 
-def main(benchmark : bool = False,
-         output_dir: str = STR_NONE,
-         prng_seed: int = INT_NONE,
-         num_nodes: int = INT_NONE,
-         num_topics: int = INT_NONE,
+def main(ctx: typer.Context, benchmark : bool = False,
```

Comment:
One question, why do we need the context here as parameter now?

Reply:
We don't, not at present; edited it out.